src/components/core/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  ReactNode,
+  RefObject,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import styled from 'styled-components';
+import TextButton from '../Button/TextButton';
+import colors from '../colors';
+
+type Props = {
+  mainText: string;
+  children?: ReactNode;
+};
+
+const refContainsEventTarget = (ref: RefObject<HTMLDivElement>, event: any) => {
+  return ref.current && ref.current.contains(event.target);
+};
+
+function Dropdown({ mainText, children }: Props) {
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const handleClick = useCallback(() => {
+    setIsDropdownVisible(!isDropdownVisible);
+  }, [isDropdownVisible]);
+
+  const dropdownMenuRef = useRef<HTMLDivElement>(null);
+  const dropdownButtonRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    function handleDocumentClick(event: any) {
+      if (!isDropdownVisible) {
+        return;
+      }
+
+      const clickedOutsideMenu = !refContainsEventTarget(
+        dropdownMenuRef,
+        event
+      );
+      const clickedOnButton = refContainsEventTarget(dropdownButtonRef, event);
+
+      if (clickedOutsideMenu && !clickedOnButton) {
+        setIsDropdownVisible(false);
+      }
+    }
+
+    // we only need to add document listener if dropdown is open
+    if (isDropdownVisible) {
```

Comment:
don't think we need this if statement if we have something similar on L34. probably safe to always register `handleDocumentClick`